File: api/tests/receiver.js
#!/usr/bin/env node
// npm install amqplib

let amqp = require('amqplib/callback_api');

amqp.connect('amqp://localhost', function(error0, connection) {
    if (error0) throw error0;

    connection.createChannel(function(error1, channel) {
        if (error1) throw error1;

        let queue = 'message_queue';

        channel.assertQueue(queue, { durable: false });

        console.log(" [*] Esperando a mensagem em %s. Aperte CTRL+C para sair", queue);

        channel.consume(queue, function(msg) {
            let body = JSON.parse(msg.content);
            console.log(" [x] Objeto encontrado: Distância = %s e Posição = %s", body.distance, body.direction)
        });
    });
});

File: api/tests/receiver.py
import pika
import json

IP = "18.228.137.154"

connection = pika.BlockingConnection(pika.ConnectionParameters(host=IP))
channel = connection.channel()

channel.queue_declare(queue='message_queue')

def callback(ch, method, properties, body):
    body = json.loads(body.decode('utf-8'))
    print(" [x] Objeto encontrado: Distância = %s e Posição = %s" % (body['distance'], body['direction']))

channel.basic_consume(
    queue='message_queue',
    auto_ack=True,
    on_message_callback=callback
)

print(' [*] Waiting for messages. To exit press CTRL+C')
channel.start_consuming()
